net/captivedetection/captivedetection.go

```diff
@@ -214,6 +214,7 @@ func (d *Detector) dialContext(ctx context.Context, network, addr string) (net.C
 	ifIndex := d.currIfIndex
 
 	dl := net.Dialer{
+		Timeout: Timeout,
```

Comment:
this is fine, but I'd also bound the context here:

```go
func (d *Detector) verifyCaptivePortalEndpoint(ctx context.Context, e Endpoint, ifIndex int) (found bool, err error) {
	req, err := http.NewRequestWithContext(ctx, "GET", e.URL.String(), nil)
```

add a:

```go
     ctx, cancel := context.WithTimeout(ctx, Timeout)
     defer cancel()
```

Reply:
done